### PR TITLE
PR for Issue #143

### PR DESCRIPTION
**Pull request by amychen108**
**1. PR Reference:** 
Completing issue #143
**2. What was changed:**
Creating the guided tour for the maps page on the website. It should be triggered automatically on first time the user logs in. There are 5 popups that happen one after another when the user clicks the next button.
**3. Why it was changed:**
This change was needed to give the user a smooth tutorial experience which shows each of the map pages features before letting the user interact freely with the website. 
**4. How it was changed:**
- Installed and added intro.js into the packages.json
- Added the intro.js to the maps/page.tsx where it is set up and started. 
- Created each of the intro popups the search-bar, notes-list, create-a-note-button, and navbar-logout.
- Created test cases for each of these popups in the introjs.test.tsx

### CODE REVIEW

File: app/lib/pages/map/page.tsx [hunk 1]
```diff
@@ -64,8 +66,32 @@ const Page = () => {
...
+            //steps using refs
+            element: noteRefs.current?.current, 
+            intro: "Welcome! Lets explore the website together"
+          },
+          {
+            element: noteRefs.current?.current, 
+            intro: "Now, lets check out how to create a note!"
+          }
+        ],
+        showProgress: true,  // Option to show progress bar
```
Comment (jwang-101) [inline app/lib/pages/map/page.tsx:89]: Unnecessary comments.
The boolean makes it clear what's happening.

File: app/lib/pages/map/page.tsx [hunk 2]
```diff
@@ -64,8 +66,32 @@ const Page = () => {
...
+        scrollToElement: true,  // Automatically scroll to element
```
Comment (jwang-101) [inline app/lib/pages/map/page.tsx:90]: Unnecessary comments #2
The boolean makes it clear what's happening.

File: app/lib/pages/map/page.tsx [hunk 3]
```diff
@@ -64,8 +66,32 @@ const Page = () => {
...
+      });
+      
+      intro.start();  // Start thme intro tour
```
Comment (jwang-101) [inline app/lib/pages/map/page.tsx:93]: Unnecessary comment no.3

**Review by jwang-101 — COMMENTED** [review]
Not much to say. Mostly looks good. I just pointed out some comment things.

**Comment by kate-holdener** [comment]
> This issue is a WIP
> 
> 1. Working on and/or completing issue [Design The Tour: Create guide for add new notes button #143](https://github.com/oss-slu/lrda_website/issues/143)
> 2. Creating the guided tour for "Create a note" tab on the website. It should be triggered automatically when pressing the next button on the welcome guide.
>    **9/15:** Created the beginning of the guided tour, however it is not referencing the correct place on the website.
>    **9/16:** Moved the guided tour to the maps page which is the page right after login.
> 3. This change was needed to give the user a smooth tutorial experience which shows each of the websites features before letting the user interact freely with the website.
> 4. **9/15:** installed and added intro.js into the packages.json
>    Added the intro.js to the WelcomePage.tsx where it is set up and started.
>    **9/16:** moved the code I wrote for the intro.js popup from the WelcomePage.tsx to the map page.tsx. And setup the intro

@amychen108 Please provide the required information in the pull request. Don't give chronological (date by date) updates. Instead, specify the information about this work, as I described in class.

File: README.md [hunk 4]
```diff
@@ -37,3 +37,4 @@ Check out our [Next.js deployment documentation](https://nextjs.org/docs/deploym
 [mui-tiptap](https://www.npmjs.com/package/mui-tiptap): A customizable Material UI styled WYSIWYG rich text editor, using Tiptap.
 
 [shadcn](https://ui.shadcn.com/) : A beautifully designed UI Component Library.
+
```
Comment (kate-holdener) [inline README.md:40]: Please don't commit unnecessary changes

File: app/lib/pages/map/page.tsx [hunk 5]
```diff
@@ -2,13 +2,16 @@
...
 import SearchBarMap from "../../components/search_bar_map";
-import { Note } from "@/app/types";
+import { Note, newNote } from "@/app/types";
 import ApiService from "../../utils/api_service";
 import DataConversion from "../../utils/data_conversion";
 import { User } from "../../models/user_class";
 import ClickableNote from "../../components/click_note_card";
 import { Switch } from "@/components/ui/switch";
 import { Skeleton } from "@/components/ui/skeleton";
+// first import introJs and introjs.css from library
```
Comment (kate-holdener) [inline app/lib/pages/map/page.tsx:12]: Remove unnecessary comments

File: pnpm-lock.yaml
[the remark this answers is not in the document]
Comment (yashb196) [reply pnpm-lock.yaml:1]: the lock file auto updates with the packages I can get those off before we push to main

**Review by BaloneyBoy97 — APPROVED** [review]

**Review by BaloneyBoy97 — APPROVED** [review]

**Review by yashb196 — APPROVED** [review]
looks good approving PR

**Comment by BaloneyBoy97** [comment]
merged it to main
